state/application.go

```diff
@@ -1005,6 +1005,60 @@ func (a *Application) SetCharm(cfg SetCharmConfig) (err error) {
 	return nil
 }
 
+// UpdateApplicationSeries updates the series for the Application.
+func (a *Application) UpdateApplicationSeries(series string) (err error) {
+	// Make sure no units are added or removed while the update-series
+	// transaction is being executed. To cover the case where a new subordinate
+	// may be added.
+	units, err := a.AllUnits()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	var unitOps []txn.Op
+	for _, u := range units {
+		subNames := u.SubordinateNames()
+		unitOps = append(unitOps, txn.Op{
+			C:      unitsC,
+			Id:     u.doc.DocID,
+			Assert: bson.D{{"subordinates", subNames}},
```

Comment:
If this sufficient? Maybe it is! I thought maybe it would have been a bit more complicated, but I am probably wrong. A test to check all this is needed to be sure (using the txn hook helpers).